### PR TITLE
[azservicebus] Fixing the ATOM surface to match the current track 2 SDKs

### PR DESCRIPTION
**Pull request by richardpark-msft**
Fixing the ATOM surface to be more consistent with how we're doing the other modern Go HTTP-based libraries:
* ISO8601 durations are represented as their strings, rather than as time.Duration, as they were before.
* Pagers use `runtime.Pager`
* Remove the Response/Result types in favor of just having a Response type.

### CODE REVIEW

**Comment by richardpark-msft** [comment]
@jhendrixMSFT  - ignore the admin_client_subscription.go for now, I apparently somehow missed adding that file into my changeset. Topics and Queues should be correct, however.

**Comment by richardpark-msft** [comment]
@jhendrixMSFT, okay, I think I've got it right now. I'm using runtime.Pager, which cleaned things up nicely.

**Comment by richardpark-msft** [comment]
@seankane-msft, I think you've been through this all enough that you'll see if I'm violating standards.

File: sdk/messaging/azservicebus/CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,43 @@
...
+    ```go
+    // new code
+    for queuePager.More() {
+		  page, err := queuePager.NextPage(context.TODO())
+
+		  if err != nil {
+			  panic(err)
+		  }
+
+		  for _, queue := range page.Items {
```
Comment (seankane-msft) [inline sdk/messaging/azservicebus/CHANGELOG.md:48]: Can we change the name of the field from Items to Queues?

File: sdk/messaging/azservicebus/admin/admin_client_queue.go [hunk 2]
```diff
@@ -161,32 +142,21 @@ type UpdateQueueOptions struct {
 }
 
 // UpdateQueue updates an existing queue.
-func (ac *Client) UpdateQueue(ctx context.Context, queueName string, properties QueueProperties, options *UpdateQueueOptions) (*UpdateQueueResponse, error) {
-	newProps, resp, err := ac.createOrUpdateQueueImpl(ctx, queueName, &properties, false)
+func (ac *Client) UpdateQueue(ctx context.Context, queueName string, properties QueueProperties, options *UpdateQueueOptions) (UpdateQueueResponse, error) {
```
Comment (seankane-msft) [inline sdk/messaging/azservicebus/admin/admin_client_queue.go:145]: I ran into this issue with KV. If a customer uses `CreateQueue`, they cannot use the embedded `QueueProperties` as an input in this `UpdateQueue` method. Sending an e-mail to discuss not embedding some of these models.

File: sdk/messaging/azservicebus/admin/admin_client_queue.go [hunk 3]
```diff
@@ -208,31 +178,20 @@ func (ac *Client) GetQueue(ctx context.Context, queueName string, options *GetQu
...
-		RawResponse: resp,
-		GetQueueResult: GetQueueResult{
-			QueueProperties: *props,
-		},
+		QueueProperties: queueItem.QueueProperties,
 	}, nil
 }
 
-// GetQueueRuntimePropertiesResult contains the result for Client.GetQueueRuntimeProperties
-type GetQueueRuntimePropertiesResult struct {
```
Comment (seankane-msft) [inline sdk/messaging/azservicebus/admin/admin_client_queue.go:226]: Changelog entry for removing all the Result structs

File: sdk/messaging/azservicebus/admin/admin_client_queue.go [hunk 4]
```diff
@@ -291,15 +246,9 @@ type ListQueuesOptions struct {
...
-// ListQueuesResult contains the result for QueuePager.PageResponse
-type ListQueuesResult struct {
-	Items []*QueueItem
-}
-
 // ListQueuesResponse contains the response fields for QueuePager.PageResponse
 type ListQueuesResponse struct {
-	ListQueuesResult
-	RawResponse *http.Response
+	Items []QueueItem
```
Comment (seankane-msft) [inline sdk/messaging/azservicebus/admin/admin_client_queue.go:251]: `Queues []QueueItem` ?

File: sdk/messaging/azservicebus/admin/admin_client_queue.go [hunk 5]
```diff
@@ -329,8 +298,7 @@ type ListQueuesRuntimePropertiesOptions struct {
 
 // ListQueuesRuntimePropertiesResponse contains the page response for QueueRuntimePropertiesPager.PageResponse
 type ListQueuesRuntimePropertiesResponse struct {
-	Items       []*QueueRuntimePropertiesItem
-	RawResponse *http.Response
+	Items []QueueRuntimePropertiesItem
```
Comment (seankane-msft) [inline sdk/messaging/azservicebus/admin/admin_client_queue.go:301]: `QueueRuntimeProperties []QueueRuntimePropertiesItem`

File: sdk/messaging/azservicebus/admin/admin_client_subscription.go [hunk 6]
```diff
@@ -247,79 +215,42 @@ type SubscriptionPropertiesItem struct {
...
-			SubscriptionName:       env.Title,
-			SubscriptionProperties: *props,
-		})
-	}
-
-	return &ListSubscriptionsResponse{
-		RawResponse: resp,
-		Items:       all,
-	}, nil
+	Items []SubscriptionPropertiesItem
```
Comment (seankane-msft) [inline sdk/messaging/azservicebus/admin/admin_client_subscription.go:218]: `SubscriptionProperties []SubscriptionPropertiesItem`

File: sdk/messaging/azservicebus/admin/admin_client_queue.go (hunk 2 above)
Comment (richardpark-msft) [reply sdk/messaging/azservicebus/admin/admin_client_queue.go:145]: Why can't they use QueueProperties as input?

Comment (richardpark-msft) [reply sdk/messaging/azservicebus/admin/admin_client_queue.go:145]: (it's an embedded property of GetQueueResponse)

```go
type GetQueueResponse struct {
	QueueProperties
}
```

Comment (seankane-msft) [reply sdk/messaging/azservicebus/admin/admin_client_queue.go:145]: Ignore my comment, you're in the clear here

Comment (richardpark-msft) [reply sdk/messaging/azservicebus/admin/admin_client_queue.go:145]: Framing this for use in the future.

File: sdk/messaging/azservicebus/admin/admin_client_queue.go (hunk 3 above)
Comment (richardpark-msft) [reply sdk/messaging/azservicebus/admin/admin_client_queue.go:226]: Oof, good point. I swore I had it in there but I DO NOT.

File: sdk/messaging/azservicebus/CHANGELOG.md (hunk 1 above)
Comment (richardpark-msft) [reply sdk/messaging/azservicebus/CHANGELOG.md]: Done.

File: sdk/messaging/azservicebus/admin/admin_client_queue.go (hunk 4 above)
Comment (richardpark-msft) [reply sdk/messaging/azservicebus/admin/admin_client_queue.go]: Done.

File: sdk/messaging/azservicebus/admin/admin_client_queue.go (hunk 3 above)
Comment (richardpark-msft) [reply sdk/messaging/azservicebus/admin/admin_client_queue.go:226]: Added.

File: sdk/messaging/azservicebus/admin/admin_client_subscription.go (hunk 6 above)
Comment (richardpark-msft) [reply sdk/messaging/azservicebus/admin/admin_client_subscription.go]: Done.

File: sdk/messaging/azservicebus/admin/admin_client_queue.go (hunk 5 above)
Comment (richardpark-msft) [reply sdk/messaging/azservicebus/admin/admin_client_queue.go]: Done. Although I did think QueueRuntimeProperties is kind of a funny name but it seemed okay in context.

**Comment by richardpark-msft** [comment]
(updated to use the just-released azcore package)

**Review by seankane-msft — APPROVED** [review]

**Pull request merged**